avrohugger-core/src/test/scala/util/AvrohuggerSpec.scala

```diff
@@ -0,0 +1,101 @@
+package util
+
+import avrohugger._
+import avrohugger.format._
+import avrohugger.format.abstractions.SourceFormat
+
+import java.io.File
+import java.nio.file.{FileSystems, Path}
+
+import org.specs2.SpecificationLike
+import org.specs2.matcher.{Matcher, Matchers, ShouldExpectable}
+
+import scala.io.Source
+
+class AvrohuggerSpec(
+  inPath: Path,
+  val outputFiles: Seq[Path],
+  sourceFormat: SourceFormat
+) extends Matchers {
+  implicit class PathExtensions(
+    path: Path
+  ) {
+    def ++(next: String) = path.resolve(next)
+    def ++(other: Path) = path.resolve(other)
+  }
+
+  val sourceFormatName = sourceFormat match {
+    case SpecificRecord => "specific"
+    case Standard => "standard"
+    case Scavro => "scavro"
+  }
+
+  val gen = new Generator(sourceFormat)
+  val inputPath = {
+    val sourceBase = FileSystems.getDefault.getPath("avrohugger-core", "src", "test", "avro")
+    (sourceBase ++ inPath)
+  }
+  val inputFile = inputPath.toFile
+  val outDir = gen.defaultOutputDir + s"/$sourceFormatName/"
+
+  private def readFile(f: File): String = {
+    val source = Source.fromFile(f)
+    try source.mkString finally source.close()
+  }
+
+  val expectedBase = FileSystems.getDefault.getPath("avrohugger-core", "src", "test", "expected", sourceFormatName)
+  val generatedBase = FileSystems.getDefault.getPath("target", "generated-sources", sourceFormatName)
+
+  private def prefixedFileString(prefixPath: Path, p: Path) = {
```

Comment:
@julianpeeters can you sanity check this quickly though? Basically, it will look for `<expected>/path/to/Something.scala` if using `Standard`, or `Specific`, but look for `<expected>/path/to/model/Something.scala` if using `Scavro`. I've never used the scavro generator before, but this appeared to be how it worked. Since I didn't know how it works though, a quick sanity check if for sure necessary.